lib/solr/testing.rb

```diff
@@ -1,19 +1,19 @@
 module Solr
   module Testing
     class << self
-      attr_reader :last_solr_request_params, :last_solr_response
-
-      def subscribe_to_events
-        if defined? ActiveSupport::Notifications
-          ActiveSupport::Notifications.subscribe('__testing_request_response.solrb') do |*args|
-            event = ActiveSupport::Notifications::Event.new(*args)
-            @last_solr_request_params = event.payload.dig(:request, :params)
-            @last_solr_response = event.payload[:response]
-          end
-        end
-      end
+      attr_accessor :last_solr_request_params
     end
   end
 end
 
-Solr::Testing.subscribe_to_events
+module Solr::Query::Request::RunnerExtension
+  def run
+    response = super
+    Solr::Testing.last_solr_request_params = request_params[:params]
+    response
+  end
+end
+
+class Solr::Query::Request::Runner
+  prepend Solr::Query::Request::RunnerExtension
```

Comment:
I didn't know about prepend, what does it do?

Reply:
can you please update the readme with an example of how to use it?

Reply:
The difference between `prepend` and `include` is that `prepend` will add the module before the original class on the ancestors chain.
The basic idea is to include my extension in a way that the extension's `run` method is found by ruby before the original `run` method. This allows me to still access the original method using `super` and do the additional stuff (storing request params for example) without having to duplicate the original method

Sure, I'll add a section to the readme.

Reply:
that's cool, thanks

Reply:
this post explains better: http://leohetsch.com/include-vs-prepend-vs-extend/

Reply:
@valve, I’ve already used perpend on machinio, as I remember